packages/eslint-plugin/tests/rules/prefer-promise-reject-errors.test.ts

```diff
@@ -0,0 +1,233 @@
+    `,
+    'new Promise((resolve, reject) => resolve(5, reject));',
+    `
+      class C {
+        #error: Error;
+        foo() {
+          Promise.reject(this.#error);
+        }
+      }
+    `,
+    `
+      const foo = Promise;
+      new foo((resolve, reject) => reject(new Error()));
+    `,
+  ],
+  invalid: [
+    'Promise.reject(5)',
+    "Promise.reject('foo')",
+    'Promise.reject(`foo`)',
+    "Promise.reject('foo', somethingElse)",
+    'Promise.reject(false)',
+    'Promise.reject(void `foo`)',
+    'Promise.reject()',
+    'Promise.reject(undefined)',
+    {
+      code: 'Promise.reject(undefined)',
+      options: [{ allowEmptyReject: true }] as const,
+    },
+    'Promise.reject(null)',
+    'Promise.reject({ foo: 1 })',
+    'Promise.reject([1, 2, 3])',
+    `
+      declare const foo: Error | undefined
+      Promise.reject(foo)
+    `,
+    `
+      declare const foo: () => Promise<string>
+      Promise.reject(await foo())
+    `,
+    `
+      declare const foo: boolean
+      Promise.reject(foo && new Error())
+    `,
+    `
+      const foo = Promise
+      foo.reject()
+    `,
+
+    'Promise.reject?.(5)',
+    'Promise?.reject(5)',
+    'Promise?.reject?.(5)',
+    '(Promise?.reject)(5)',
+    '(Promise?.reject)?.(5)',
+    `Promise['reject'](5)`,
+
+    // Assignments with mathematical operators will either evaluate to a primitive value or throw a TypeError
+    'Promise.reject(foo += new Error())',
+    'Promise.reject(foo -= new Error())',
+    'Promise.reject(foo **= new Error())',
+    'Promise.reject(foo <<= new Error())',
+    'Promise.reject(foo |= new Error())',
+    'Promise.reject(foo &= new Error())',
+
+    'new Promise(function(resolve, reject) { reject() })',
+    'new Promise(function(resolve, reject) { reject(5) })',
+    'new Promise((resolve, reject) => { reject() })',
+    'new Promise((resolve, reject) => reject(5))',
+    `
+      new Promise((resolve, reject) => {
+        fs.readFile('foo.txt', (err, file) => {
+          if (err) reject('File not found')
+          else resolve(file)
+        })
+      })
+    `,
+    'new Promise((yes, no) => no(5))',
+    'new Promise(({foo, bar, baz}, reject) => reject(5))',
+    'new Promise(function(reject, reject) { reject(5) })',
+    'new Promise(function(foo, arguments) { arguments(5) })',
+    'new Promise((foo, arguments) => arguments(5))',
+    'new Promise(function({}, reject) { reject(5) })',
+    'new Promise(({}, reject) => reject(5))',
+    'new Promise((resolve, reject, somethingElse = reject(5)) => {})',
+    `
+      declare const foo: {
+        bar: PromiseConstructor
+      }
+      new foo.bar((resolve, reject) => reject(5))
+    `,
+    `
+      declare const foo: {
+        bar: PromiseConstructor
+      }
+      new (foo?.bar)((resolve, reject) => reject(5))
+    `,
+    `
+      const foo = Promise
+      new foo((resolve, reject) => reject(5))
+    `,
+  ].map<TSESLint.InvalidTestCase<MessageIds, Options>>(invalidCase => {
```

Comment:
[Testing] We've been moving away from auto-generated tests. They have a habit of being imprecise and hard to statically analyze. We'd also want to be able to customize/see more directly what line & column ranges the errors would be reported on. -1 from me on the `.map`, can you switch to direct object literals please?